Apps/JobScheduler/src/Jobs/CommunicationJob.cs

```diff
@@ -0,0 +1,155 @@
+        private readonly IEmailQueueService emailQueueService;
+        private readonly GatewayDbContext dbContext;
+
+        private readonly int maxFetchSize;
+        private readonly string fromEmailAddressHGDonotreply;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CommunicationJob"/> class.
+        /// </summary>
+        /// <param name="configuration">The configuration to use.</param>
+        /// <param name="logger">The logger to use.</param>
+        /// <param name="communicationDelegate">The Communication delegate to use.</param>
+        /// <param name="commEmailDelegate">The Communication Email delegate to use.</param>
+        /// <param name="emailQueueService">The email queue service to use.</param>
+        /// <param name="dbContext">The db context to use.</param>
+        public CommunicationJob(IConfiguration configuration, ILogger<CommunicationJob> logger, ICommunicationDelegate communicationDelegate, ICommunicationEmailDelegate commEmailDelegate, IEmailQueueService emailQueueService, GatewayDbContext dbContext)
+        {
+            Contract.Requires((configuration != null) && (emailQueueService != null));
+            this.configuration = configuration!;
+            this.logger = logger;
+            this.communicationDelegate = communicationDelegate!;
+            this.commEmailDelegate = commEmailDelegate!;
+            this.emailQueueService = emailQueueService!;
+            this.dbContext = dbContext;
+
+            IConfigurationSection commEmailJobSection = this.configuration.GetSection("CreateCommEmailsForNewCommunications");
+            this.fromEmailAddressHGDonotreply = commEmailJobSection.GetValue<string>("FromEmailAddressHGDonotreply");
+            this.maxFetchSize = commEmailJobSection.GetValue<int>("MaxFetchSize", 250);
+        }
+
+        /// <inheritdoc />
+        [DisableConcurrentExecution(ConcurrencyTimeout)]
+        public void CreateCommunicationEmailsForNewCommunications()
+        {
+            this.logger.LogDebug($"Creating emails & communication emails...");
+            List<Communication> communications = this.communicationDelegate.GetEmailCommunicationsToSend();
+
+            if (communications.Count > 0)
+            {
+                this.logger.LogInformation($"Found {communications.Count} communications which need to be processed / re-processed.");
+
+                foreach (Communication communication in communications)
+                {
+#pragma warning disable CA1031 //We want to catch exception.
+                    try
+                    {
+                        List<UserProfile> usersToSendCommEmails = new List<UserProfile>();
+                        bool moreUsersToCreateCommunicationEmails = false;
+                        do
+                        {
+                            usersToSendCommEmails = this.commEmailDelegate.GetActiveUserProfilesByCommunicationId(communication.Id, this.maxFetchSize);
```

Comment:
The method should be more descriptive like getProfilesForCommunication

Reply:
renaming..